File: Pattern41.c
#include<stdio.h>

void main() {
    int n, row, col;
    printf("Enter n: ");
    scanf("%d",&n);

    for(int i=1; i<=n; i++) {
            for(int j=i; j<=n; j++){
                printf("%d", j);
                printf(" ");
            }

           printf("\n");  

           if(i == n)
                break;

           for(int k=1; k<=i; k++)
                printf(" ");
        }
        
        for(int i=1; i<=n-2; i++)
            printf(" ");

        for(int i=1; i<n; i++) {
            for(int j=n-i; j<=n; j++){
                printf("%d", j);
                printf(" ");
            }

           printf("\n");
            
           for(int k=1; k<n-i-1; k++)
                printf(" ");
        }
}

File: PATTERNS/Pattern41/Pattern41.go
package main

import "fmt"

func main() {

	var n int
	fmt.Printf("Enter the row number:")
	fmt.Scanln(&n)
	var flag int

	for i := 1; i <= 2*n-1; i++ {
		if i <= n {
			flag = i
		} else {
			flag = 2*n - i
		}
		var flag1 int = n
		for k := 1; k <= flag; k++ {
			fmt.Printf(" ")
		}
		for j := flag; j <= flag1; j++ {
			fmt.Printf("%d ",j)
		}
		fmt.Printf("\n")
	}
}
